### DISCUSSION TITLE
Use 'useMutation' in a helper function in my case

### DISCUSSION THREAD

**Discussion by Xi-gofore** [TanStack/query#1904] [General]
I have a javascript file called _helper.js_, inside it is a bunch of helper functions.

```
const calculateData = () => {
     ...
}

const updateDataIfNeeded = async ({myData}) => {
     if(myData.length>10) {
       // I need to fire a POST request here
     }
}
```

As you can see above I need to fire a POST request in the helper function `updateDataIfNeeded` inside a `if` condition. The problem is that I am not sure how can I take advantage of the `useMutation()` hook to fire the request since the _helper.js_ is neither a React function component nor a custom React Hook function. 

I wonder what is the best practice to fulfil my need and use `useMutation()` for the POST request in my helper function?

**Comment by TkDodo** [comment]
you can pass the mutate function from `useMutation` to it, or call useMutation in the helper itself it it's a hook:

```
const updateDataIfNeeded = ({mutate, myData}) => {
     if(myData.length>10) {
       mutate(myData)
     }
}

function MyComponent() {
    const { mutate } = useMutation(..)
    updateDataIfneeded({ mutate, myData })
```

```
const useUpdateDataIfNeeded = ({myData}) => {
    const { mutate } = useMutation(..) 
     if(myData.length>10) {
       mutate(myData)
     }
}
```

**Reply by Xi-gofore** [reply]
In your answer, what is that `useUpdateDataIfNeeded ` ? Is that a custom hook? Where do you use it? It is confusion in your answer.
